modules/network/src/main/java/org/apache/ignite/internal/network/recovery/RecoveryClientHandshakeManager.java

```diff
@@ -237,17 +230,16 @@ private void onHandshakeStartMessage(HandshakeStartMessage handshakeStartMessage
             return;
         }
 
-        this.remoteLaunchId = handshakeStartMessage.launchId();
-        this.remoteConsistentId = handshakeStartMessage.consistentId();
+        this.remoteNode = handshakeStartMessage.serverNode().asClusterNode();
 
-        ChannelKey channelKey = new ChannelKey(remoteConsistentId, remoteLaunchId, connectionId);
+        ChannelKey channelKey = new ChannelKey(remoteNode.name(), UUID.fromString(remoteNode.id()), connectionId);
```

Comment:
Should we pass the cluster node to all such places?

Reply:
`ChannelKey` is defined as a triple `<consistentId, launchId, channelIndex>`. According to the `ClusterNodeImpl` code, it considers 2 nodes equal if their name and address match. The ID is not even taken into consideration (which is weird, by the way). So with the current code replacement of `<node, channelIndex>` will just break everything.

Even if it did not break it (due to having a compatible equals semantics), it seems more straightforward and safer to list the components explicitly.